tests/integration/__tests__/cache.js

```diff
@@ -0,0 +1,327 @@
+        ])
+      );
+
+      // Update `a.js`
+      await fs.writeFile(path.join(dir, "a.js"), "const a = `a`;");
+
+      const { stdout: secondStdout } = await runPrettier(dir, [
+        "--cache",
+        "--cache-strategy",
+        "content",
+        "--write",
+        ".",
+      ]);
+      expect(stripAnsi(secondStdout).split("\n").filter(Boolean)).toEqual(
+        // the cache of `b.js` is only available.
+        expect.arrayContaining([
+          expect.stringMatching(/^a\.js .+ms$/),
+          expect.stringMatching(/^b\.js .+ms \(cached\)$/),
+        ])
+      );
+    });
+
+    it("does'nt re-format when timestamp has been updated", async () => {
+      const { stdout: firstStdout } = await runPrettier(dir, [
+        "--cache",
+        "--cache-strategy",
+        "content",
+        "--write",
+        ".",
+      ]);
+      expect(stripAnsi(firstStdout).split("\n").filter(Boolean)).toEqual(
+        expect.arrayContaining([
+          expect.stringMatching(/^a\.js .+ms$/),
+          expect.stringMatching(/^b\.js .+ms$/),
+        ])
+      );
+
+      // update timestamp
+      const time = new Date();
+      await fs.utimes(path.join(dir, "a.js"), time, time);
+
+      const { stdout: secondStdout } = await runPrettier(dir, [
+        "--cache",
+        "--cache-strategy",
+        "content",
+        "--write",
+        ".",
+      ]);
+      expect(stripAnsi(secondStdout).split("\n").filter(Boolean)).toEqual(
+        expect.arrayContaining([
+          expect.stringMatching(/^a\.js .+ms \(cached\)$/),
+          expect.stringMatching(/^b\.js .+ms \(cached\)$/),
+        ])
+      );
+    });
+
+    it("re-formats when options has been updated.", async () => {
+      const { stdout: firstStdout } = await runPrettier(dir, [
+        "--cache",
+        "--cache-strategy",
+        "content",
+        "--write",
+        ".",
+      ]);
+      expect(stripAnsi(firstStdout).split("\n").filter(Boolean)).toEqual(
+        expect.arrayContaining([
+          expect.stringMatching(/^a\.js .+ms$/),
+          expect.stringMatching(/^b\.js .+ms$/),
+        ])
+      );
+
+      const { stdout: secondStdout } = await runPrettier(dir, [
+        "--cache",
+        "--write",
+        "--cache-strategy",
+        "content",
+        "--trailing-comma",
+        "all",
+        ".",
+      ]);
+      expect(stripAnsi(secondStdout).split("\n").filter(Boolean)).toEqual(
+        expect.arrayContaining([
+          expect.stringMatching(/^a\.js .+ms$/),
+          expect.stringMatching(/^b\.js .+ms$/),
+        ])
+      );
+    });
+
+    it("removes cache file when run Prettier without `--cache` option", async () => {
+      await runPrettier(dir, ["--cache", "--write", "."]);
+      await expect(fs.stat(defaultCacheFile)).resolves.not.toThrowError();
+      await runPrettier(dir, ["--write", "."]);
+      await expect(fs.stat(defaultCacheFile)).rejects.toThrowError();
+    });
+  });
+});
```

Comment:
Can we add a test for stdin? So we won't accidentally use cache when formatting stdin.

Reply:
[9cf4a31](https://github.com/prettier/prettier/pull/12800/commits/9cf4a312aafa1037d48d37f034c857018aa3aeb4)